services/ymax-planner/src/support.ts

```diff
@@ -238,11 +286,37 @@ const findBlockByTimestamp = async (
 export const buildTimeWindow = async (
   provider: JsonRpcProvider,
   publishTimeMs: number,
+  log: (...args: unknown[]) => void,
+  chainId: CaipChainId,
   fudgeFactorMs = 5 * 60 * 1000, // 5 minutes to account for cross-chain clock differences
 ) => {
```

Comment:
This collection of arguments should now have some names.
```suggestion
export const buildTimeWindow = async (
  chainId: CaipChainId,
  provider: JsonRpcProvider,
  publishTimeMs: number,
  {
    log = () => {},
    fudgeFactorMs = 5 * 60 * 1000, // 5 minutes to account for cross-chain clock differences
  }: { log?: (...args: unknown[]) => void, fudgeFactorMs?: number }
) => {
```
or better yet (since `chainId` is only used to call `getBlockTimeMs`)
```suggestion
export const buildTimeWindow = async (
  provider: JsonRpcProvider,
  publishTimeMs: number,
  {
    meanBlockDurationMs = 12_000, // Default to Ethereum's conservative 12s
    log = () => {},
    fudgeFactorMs = 5 * 60 * 1000, // 5 minutes to account for cross-chain clock differences
  }: {
    meanBlockDurationMs?: number,
    log?: (...args: unknown[]) => void,
    fudgeFactorMs?: number,
  }
) => {
```
(moving block duration estimates to be a caller responsibility)